Libplanet.Benchmarks/Store.cs

```diff
@@ -0,0 +1,147 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using BenchmarkDotNet.Attributes;
+using Libplanet.Blocks;
+using Libplanet.Crypto;
+using Libplanet.Store;
+using Libplanet.Tests;
+using Libplanet.Tests.Common.Action;
+using Libplanet.Tests.Store;
+using Libplanet.Tx;
+
+namespace Libplanet.Benchmarks
+{
+    public class Store
+    {
+        private static readonly ImmutableArray<Block<DumbAction>> Blocks;
+        private static readonly int BlockCount;
+        private static readonly ImmutableArray<Transaction<DumbAction>> Txs;
+        private static readonly int TxsCount;
+        private IStore _store = null;
+
+        static Store()
+        {
+            List<Block<DumbAction>> blocks = new List<Block<DumbAction>>();
+            List<Transaction<DumbAction>> txs = new List<Transaction<DumbAction>>();
+            Block<DumbAction> block = TestUtils.MineGenesis<DumbAction>();
+            blocks.Add(block);
+            PrivateKey key = new PrivateKey();
+            long nonce = 0;
+            for (int i = 0; i < 500; i++)
+            {
+                List<Transaction<DumbAction>> blockTxs = new List<Transaction<DumbAction>>();
+                for (int j = 0; j < i % 5; j++)
+                {
+                    blockTxs.Add(Transaction<DumbAction>.Create(nonce++, key, new DumbAction[0]));
+                }
+                block = TestUtils.MineNext(block, blockTxs);
+                blocks.Add(block);
+                txs.AddRange(blockTxs);
+            }
+            Blocks = blocks.ToImmutableArray();
+            BlockCount = Blocks.Length;
+            Txs = txs.ToImmutableArray();
+            TxsCount = Txs.Length;
+        }
+
+        [IterationSetup(
+            Targets = new []
+            {
+                nameof(PutFirstEmptyBlock),
+                nameof(PutFirstBlockWithTxs),
+                nameof(PutFirstTx),
+            }
+        )]
+        public void InitializeFixture()
+        {
+            var fx = new LiteDBStoreFixture();
+            _store = fx.Store;
+        }
+
+        [Benchmark]
+        public void PutFirstEmptyBlock()
+        {
+            _store.PutBlock(Blocks[0]);
+        }
+
+        [Benchmark]
+        public void PutFirstBlockWithTxs()
+        {
+            _store.PutBlock(Blocks[5]);
+        }
+
+        [IterationSetup(
+            Targets = new []
+            {
+                nameof(PutBlockOnManyBlocks),
+                nameof(GetBlock),
+            }
+        )]
+        public void PutManyBlocks()
+        {
+            InitializeFixture();
+            int i = 0;
+            foreach (Block<DumbAction> block in Blocks)
+            {
+                _store.PutBlock(block);
+                i++;
+                if (i >= Blocks.Length - 1)
```

Comment:
What is this for?

Reply:
It's intended to avoid to put the last block until this initialization phase, and the last one is for the benchmark method.